### DISCUSSION TITLE
Api rules, allow access by shared relation?

### DISCUSSION THREAD

**Discussion by snrn-Pontus** [pocketbase/pocketbase#3639] [Q&A] [answered]
I have a dogs table, that contains a field called owners, it's a multi-relationship to the users table.

Basically I want any owner of a dog to see the other owners. 

`id = @request.auth.id
|| (@collection.dogs.owners.id ?= id 
  && @collection.dogs.owners.id ?= @request.auth.id)`

Using this in the users view and list rules doesn't allow me to access users set as owners on the same dogs as me. 

Am I missing something? 

The dogs table has this rule

`@request.auth.id ?= owners.id`

I've tried a lot of different things and it either seems I can get all users or only myself, haven't been able to create a rule where I can only get users who are owners on the same dog.

**Comment by ganigeorgiev** [comment, marked as answer]
The rule `@collection.dogs.owners.id ?= id && @collection.dogs.owners.id ?= @request.auth.id` will not work because it creates a JOIN with each array value and both conditions will be applied against the same row value which can't be equal to 2 different things (aka. it will end up something like `dogs_owners_item_id = id AND dog_owners_item_id = authId`).

This can be solved if we had support for `@collection.*` aliases OR "has" operator (_both are planned for v0.20/v0.21_).

One possible workaround for the moment could be to use a loose like/contains match comparison for the first owners condition, aka. something like:
```js
id = @request.auth.id || (@collection.dogs.owners ?~ id  && @collection.dogs.owners.id ?= @request.auth.id)
```

**Reply by snrn-Pontus** [reply]
Awesome! Thanks! Is there a better way to have this relationship? Or is it just a matter of waiting for features?

**Reply by ganigeorgiev** [reply]
I think your data structure is fine, it is more of an API rule limitation at the moment.
